### ISSUE TITLE
How to close a channel when there is no network?

### ISSUE THREAD

**Issue by michalzaq12** [ably/ably-js#2147]
Ably 2.3.2

How do I close a channel when there's no network? (The channel is no longer needed, and the client's presence should also be removed).
I don't see any pattern for gracefully closing a channel in your documentation.

My code currently looks like this:
```js
channel.detach()
   .finally(() => connection.release(<channelName>))
```
When there is no network:
1. `detach()` fails with: `_ErrorInfo: Connection to server unavailable`
2. `release()` fails with: `Can only release a channel in a state where there is no possibility of further updates from the server being received (initialized, detached, or failed); was suspended`

Problems:
Once the network returns, the channel will re-attched and presence will be re-entered.
